src/server/detail/save_stages_controller.h

```diff
@@ -19,16 +17,28 @@ class Service;
 
 namespace detail {
 
+class SnapshotStorage;
+
+struct LastSaveInfo {
```

Comment:
nit: since we return it from a ServerFamily function, it can be declared outside `detail` namespace